### ISSUE TITLE
Bugs in winnie the pooh's home run derby

### ISSUE THREAD

**Issue by DependentAppearance0** [ruffle-rs/ruffle#1374]
I've been experimenting running some games in ruffle recently and this game appears to run fairly well with some quite major bugs. The game is playable to at least halfway. The swf im using I downloaded from [here](https://img.y8.com/cloud/y8-flash-game/contents/item_versions/flash_games/90536/original/winnie_the_pooh_s_home_run_derby.swf?1513694764). I've only tried this on the demo hosted on ruffle.rs on firefox 82.

Bug 1.
The ingame audio mute button isn't working. This appears to be due to Sound.setVolume being unimplemented.

Bug 2.
The ingame shop allows purchasing items even when the player has no money resulting in the player having negative money. It's also possible to buy more items than the normal limit of 20 in the game which causes the players money to go to NaN.

Bug 3.
In game the game displays distances as floating point numbers which are sometimes off by like 0.0000000000001 causing your distance to be something like 53.0000000009m causing graphical bugs as the text isn't supposed to be that long.

Apart from theese bugs the game is perfectly playable and works great.

**Comment by Herschel** [comment]
Thanks for the report!

2) Need to implement `Button.enabled`.

3) float-to-string for AS2 truncates the value to some number of decimal places (12)?

**Comment by DependentAppearance0** [comment]
The buttons in the sho work nicely now.

There are still the issues with numbers and the music mute button doesnt work because `Sound.setVolume` is unimplemented.

Here is an example of the broken numbers. They sometimes also appear during gameplay when the distance for that specific home run is shown.
![example of broken numbers](https://user-images.githubusercontent.com/68330228/101066187-be600200-359e-11eb-9319-267aacced4e3.png)

**Comment by Toad06** [comment]
`Sound.setVolume` implemented in #2825.

The issue about the broken numbers is likely related to the floating point inaccuracy:

```
var a = 0.7 - 0.5;
trace(a);
// Ruffle: 0.19999999999999996
// Flash : 0.2 (probably rounded to 0.2 when converted to a string)

a = a - 0.2;
trace(a);
// Ruffle: -5.551115123125783e-17
// Flash : -5.55111512312578e-17
```

**Comment by Toad06** [comment]
Broken numbers fixed by #6617.
